src/plugins/index_pattern_management/public/components/index_pattern_table/index_pattern_table.tsx

```diff
@@ -150,43 +128,44 @@ export const IndexPatternTable = ({ canSave, history }: Props) => {
   ];
 
   const createButton = canSave ? (
-    <CreateButton options={creationOptions}>
+    <EuiButton
+      fill={true}
+      iconType="plusInCircle"
+      onClick={() => setShowCreateDialog(true)}
+      data-test-subj="createIndexPatternButton"
+    >
       <FormattedMessage
         id="indexPatternManagement.indexPatternTable.createBtn"
         defaultMessage="Create index pattern"
       />
-    </CreateButton>
+    </EuiButton>
   ) : (
     <></>
   );
 
-  if (isLoadingSources || isLoadingIndexPatterns) {
+  if (isLoadingIndexPatterns) {
     return <></>;
   }
 
-  const hasDataIndices = sources.some(({ name }: MatchedItem) => !name.startsWith('.'));
-
-  if (!indexPatterns.length) {
-    if (!hasDataIndices && !remoteClustersExist) {
-      return (
-        <EmptyState
-          onRefresh={loadSources}
-          docLinks={docLinks}
-          navigateToApp={application.navigateToApp}
-          canSave={canSave}
-        />
-      );
-    } else {
-      return (
-        <EmptyIndexPatternPrompt
-          canSave={canSave}
-          creationOptions={creationOptions}
-          docLinksIndexPatternIntro={docLinks.links.indexPatterns.introduction}
-          setBreadcrumbs={setBreadcrumbs}
-        />
-      );
-    }
-  }
+  const displayIndexPatternEditor = showCreateDialog ? (
+    <IndexPatternEditor
+      onSave={(indexPattern) => {
+        history.push(`patterns/${indexPattern.id}`);
+      }}
+      onCancel={() => setShowCreateDialog(false)}
+      services={{
```

Comment:
Instead of exporting `IndexPatternEditor`, we could export a preconnected component on start contact, similar to how it is done in data plugin with top bar. 
in this case, API could be a bit nicer because no need to explicitly pass down services.

Reply:
addressed